src/main/webapp/shopping-results.js

```diff
@@ -0,0 +1,126 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+let photoCategory;
+let blobKeyString;
+
+/**  
+ * Gets and stores information about the photo uploaded by the user by making a GET request
+ * to '/get-image-info'. 
+ * {@code photoCategory} and {@code blobKeyString} are needed for making the GET request to 
+ * the PhotoShoppingServlet.
+ */
+async function fetchUploadedImageInfo() {
+  const response = await fetch('/get-image-info');
+  if (!response.ok) {
+    throw new Error(`HTTP error! status: ${response.status}`);
+  }
+  
+  // Get and store the photo category (i.e. product, list or barcode) and the keystring 
+  // of the blobkey.
+  let uploadedPhotoInformation = await response.text();
+
+  uploadedPhotoInformation = uploadedPhotoInformation.split('\n');
+
+  photoCategory = uploadedPhotoInformation[0];
+  blobKeyString = uploadedPhotoInformation[1];
+}
+
+/**
+ * Builds the Shopping Results Page UI by integrating product results from 
+ * Google Shopping into the webpage. 
+ */
+async function buildShoppingResultsUI() {
+  // Make a GET request to '/photo-shopping-request' to scrape the Google Shopping 
+  // search query results. The request returns the complete HTML of the SERP, stored
+  // into {@code shoppingSearchResultsPage}.
+
+  // Add needed parameters to the URL fetched.
+  const fetchURL = 
+      `/photo-shopping-request?photo-category=${photoCategory}&blob-key=${blobKeyString}`;
+
+  const response = await fetch(fetchURL).catch(handleError);
+
+  if (!response.ok) {
+    return Promise.reject(response);
+  }
```

Comment:
Um since you are catching the error in the line above, I don't think we need this anymore, do we?
We can just add it in the `handleError` method if needed.

Reply:
It is needed, as the `handleError` method (if needed to be called) returns a response. Then, at those lines, I'm checking the response and returning a promise (the function is asynchronous so it must return a Promise).

Reply:
Okay that makes sense!